### PR TITLE
Building a serverless image recognition website with ML

### PR DESCRIPTION
**Pull request by cpyle0819**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [x] The title of this pull request reflects the title of my content.
- [x] I have reviewed the [Content Review Checklist](https://github.com/build-on-aws/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [x] This pull request contains one piece of content (post or page).

Is this pull request linked to a Content Proposal or Idea Suggestion Issue? If so, please link to it below using the number or full link.

```text
Resolves #5

or

Content for https://github.com/build-on-aws/content/issues/5
```

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](/LICENSE).**

### CODE REVIEW

File: posts/cloud-journeys/01-serverless-image-recognition-app/index.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+But we wanted to push further.
+
+Because Dan was going to use this app irregularly, and only download a subset of images at any given time, how low could we get his storage costs? S3 offers a number of storage tiers. Generally speaking, accepting a higher latency for object retrieval means cheaper long-term storage. This led us to S3’s longer term storage tiers. They come with a number of trade-offs for engineers.
+
+Objects can be individually moved between storage tiers. This is done with policies (to move things to longer term storage) and batch jobs (to move things to faster access storage). Our initial plan was to use a lifecycle policy to move images to the longer term storage tiers after image labeling took place. Then, when Dan asked to retrieve data, we would start a batch job to restore those images to a temporary location (probably another S3 bucket) that would be emptied after we were done handling the download (see 4. Retrieving images).
```
Comment (darko-mesaros) [inline posts/cloud-journeys/01-serverless-image-recognition-app/index.md:68]: Add a hyperlink to the segment here.

Comment (darko-mesaros) [reply posts/cloud-journeys/01-serverless-image-recognition-app/index.md:68]: (Segment 4. Retrieving images)

File: posts/cloud-journeys/01-serverless-image-recognition-app/index.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+Each of these were important for a reason, based on Dan’s use case. More than anything, this exercise reinforced for us the importance of domain-driven design in building an application. Programs are built for people, and knowing how people will use them anchored our decisions and research in this cross-service example.
+
+The code for this example is available in a few places. The main entry point is our [CDK](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/resources/cdk/photo_asset_manager). This has all the AWS managed pieces together in one place for review. There are implementations for the Lambda functions in each of the language directories (see appendix A). And there is one implementation of the [React UI](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/resources/clients/react/elros-pam).
```
Comment (darko-mesaros) [inline posts/cloud-journeys/01-serverless-image-recognition-app/index.md:139]: The CDK Link seems to be broken.

File: posts/cloud-journeys/01-serverless-image-recognition-app/index.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+<img src="./images/labels.png"/>
+
+## Appendix A - Implementations
+
+Most of the application resources for this example are shared. The Lambda functions are implemented in different languages to showcase the use of the AWS SDK in those languages.
+
+- [Java (V2)](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/javav2/usecases/pam_source_files)
+- [Rust](https://github.com/awsdocs/aws-doc-sdk-examples/pull/4682)
+- [Front End](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/resources/clients/react/elros-pam)
+- [CDK](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/resources/cdk/photo_asset_manager)
```
Comment (darko-mesaros) [inline posts/cloud-journeys/01-serverless-image-recognition-app/index.md:159]: This one is also broken

**Comment by darko-mesaros** [comment]
I've left notes, overall the technical review feedback is good.

But something is a miss here, and this may be changed in the future elements of this series. As a reader of this article, I expected to be able to follow along or at least attempt myself to implement the same solution you did. This may be just my wrong expectations, but in that case - make sure to state in the start, that a full guide and code to achieve this will be available at the end :) 

Good work

**Comment by cpyle0819** [comment]
Hey @darko-mesaros , thanks for the review. I addressed the concerns.

**Review by lenadroid — APPROVED** [review]

**Pull request merged**